src/Bicep.LangServer/Handlers/BicepSignatureHelpHandler.cs

```diff
@@ -0,0 +1,271 @@
+                return;
+            }
+
+            if (CheckIfSignatureHelpSimilar(context.ActiveSignatureHelp, signatureHelp))
+            {
+                // the signature help is for the same function so we can reuse the active signature index
+                // this prevents resetting of the active signature when multiple overloads are ambiguous and the user selected a specific one manually
+                signatureHelp.ActiveSignature = context.ActiveSignatureHelp.ActiveSignature;
+            }
+        }
+
+        private static bool CheckIfSignatureHelpSimilar(SignatureHelp active, SignatureHelp @new)
+        {
+            // local function
+            static string GetFunctionName(SignatureInformation info)
+            {
+                var openParenIndex = info.Label.IndexOf(FunctionArgumentStart, StringComparison.Ordinal);
+                return openParenIndex <= 0 ? info.Label : info.Label.Substring(0, openParenIndex - 1);
+            }
+
+            var newSignatureCount = @new.Signatures.Count();
+            if (active.ActiveSignature > newSignatureCount || active.Signatures.Count() != newSignatureCount)
+            {
+                return false;
+            }
+
+            return active.Signatures
+                .Zip(@new.Signatures)
+                .All(tuple => string.Equals(GetFunctionName(tuple.First), GetFunctionName(tuple.Second), StringComparison.Ordinal) &&
+                              string.Equals(tuple.First.Documentation?.MarkupContent?.Value, tuple.Second.Documentation?.MarkupContent?.Value, StringComparison.Ordinal));
+        }
+
+        private static List<TypeSymbol> NormalizeArgumentTypes(ImmutableArray<FunctionArgumentSyntax> arguments, SemanticModel semanticModel)
+        {
+            return arguments
+                .Select(arg =>
+                {
+                    var argumentType = semanticModel.GetTypeInfo(arg);
+                    return argumentType is ErrorType ? LanguageConstants.Any : argumentType;
+                })
+                .ToList();
+        }
+
+        private static SignatureHelp CreateSignatureHelp(ImmutableArray<FunctionArgumentSyntax> arguments, List<TypeSymbol> normalizedArgumentTypes, FunctionSymbol symbol, int offset)
+        {
+            // exclude overloads where the specified arguments have exceeded the maximum
+            // allow count mismatches because the user may not have started typing the arguments yet
+            var matchingOverloads = symbol.Overloads
+                .Where(fo => !fo.MaximumArgumentCount.HasValue || normalizedArgumentTypes.Count <= fo.MaximumArgumentCount.Value)
+                .Select(overload => (overload, result: overload.Match(normalizedArgumentTypes, out _, out _)))
+                .ToList();
+
+            int activeSignatureIndex = matchingOverloads.IndexOf(tuple => tuple.result == FunctionMatchResult.Match);
```

Comment:
>activeSignatureIndex [](start = 16, length = 20)

I think in the future we can consider sorting the overloads by number of mismatched types descending or something like that.